### PR TITLE
Add option to override drawing with 90 degree lock by holding shift key

### PR DESCRIPTION
**Pull request by kipe**
This pull request adds an option to temporarily "unlock" drawing with 90 degree angles by holding down the shift key.

https://github.com/user-attachments/assets/24eb4fcf-b113-4a98-a7ac-04fceec38422

### CODE REVIEW

File: modules/editable-layers/src/edit-modes/draw-90degree-polygon-mode.ts [hunk 1]
```diff
@@ -29,7 +29,10 @@ export class Draw90DegreePolygonMode extends GeoJsonEditMode {
     const {mapCoords} = props.lastPointerMoveEvent;
 
     let p3;
-    if (clickSequence.length <= 1) {
+    if (clickSequence.length <= 1 ||
+      // Allow overriding 90 degree enforcement with shift key
+      (props.modeConfig && props.modeConfig.overrideWithShift && props.lastPointerMoveEvent.sourceEvent.shiftKey)
```
Comment (ibgreen) [inline modules/editable-layers/src/edit-modes/draw-90degree-polygon-mode.ts]:
```suggestion
      (props.modeConfig?.overrideWithShift && props.lastPointerMoveEvent.sourceEvent.shiftKey)
```

**Review by ibgreen — COMMENTED** [review]
Thanks for the contribution and for also taking care of the docs.

Are there any tests that can be extended or created to cover the new functionality?

**Comment by kipe** [comment]
> Thanks for the contribution and for also taking care of the docs.
> 
> Are there any tests that can be extended or created to cover the new functionality?

I'm fairly sure there is, just started using the library tonight, so haven't gotten around to look into the test system yet :)
With a fast look it seems like there isn't any tests for the 90 degree drawing, so a good first step would be to actually starting to test it.

I think it would be best to first add tests to PR #108 , as it fixes the functionality before adding new features to it (although I'll be opening another PR in a moment :P).

**Comment by ibgreen** [comment]
@kipe That sounds good. I know it can be quite a bit of extra work, but if you are able to go the extra mile and start adding tests (even where the existing tests are thing), it will greatly increase the trust level and speed when merging your PRs.

**Comment by kipe** [comment]
> @kipe That sounds good. I know it can be quite a bit of extra work, but if you are able to go the extra mile and start adding tests (even where the existing tests are thing), it will greatly increase the trust level and speed when merging your PRs.

Yeah, I'd be happy to do it as we'll most likely starting to use this library in a product. It seems great so far!

**Comment by ibgreen** [comment]
Linter fails. Run `yarn lint fix` and push again.

**Comment by ibgreen** [comment]
@kipe If you want this landed, please fix lint issues.

**Comment by ibgreen** [comment]
Closing due to inactivity. if you fix the lint issues please reopen and I will land and publish

**Pull request closed without merging**